contracts/src/BeefyClient.sol

```diff
@@ -260,7 +260,7 @@ contract BeefyClient {
             signatureUsageCount = currentValidatorSet.usageCounters.get(proof.index);
             currentValidatorSet.usageCounters.set(proof.index, signatureUsageCount.saturatingAdd(1));
             vset = currentValidatorSet;
-        } else if (commitment.validatorSetID == nextValidatorSet.id) {
+        } else if (commitment.validatorSetID >= nextValidatorSet.id) {
```

Comment:
We need some tests in the solidity that skips a mandatory commitment. 

Example scenario:
We are in validator set id 6 (next validator set 7), we skip 7 and 8 and then import 9. After importing 9 the current validator set will be 7 and the next validator set will be 10.

See here: https://github.com/Snowfork/snowbridge/pull/1215/files#diff-473fead6cd500a4a05f06d0cea4d9ef166c9bdbe2a2e1a06a33d2e1aedcc428eR379-R382

We also need to make sure that a future commit does not commit to a block number in the past and a test that covers it.

Context:
https://github.com/Snowfork/snowbridge/pull/1137#issuecomment-1923652664
https://github.com/Snowfork/snowbridge/pull/1137#issuecomment-1970136640

Reply:
Good catch Alistair, yeah, we need more unit tests for this work.

Reply:
Yeah, it's already discussed here in https://github.com/Snowfork/snowbridge/pull/1137#issuecomment-1923652664, 


> For skipping sessions, I reckon this is acceptable:
> 
> if you jump directly to session 7 while it is still the current session, then unless you remap signatures from the authorities from session 8, you'd have to wait for session 8 to send any new commitment.
> if you want to jump to session 7, just send a commitment for session 6 first

Reply:
Yeah, so in this case the current&next validatorId pair will be (7,10) and `latestBeefyBlock` is in session 9.

I would expect a minor change here
https://github.com/Snowfork/snowbridge/blob/40f95923a816184dfb39c8e1e30a89b4f8d1c0f1/contracts/src/BeefyClient.sol#L259
from ```if (commitment.validatorSetID == currentValidatorSet.id)``` to ``if (commitment.validatorSetID == currentValidatorSet.id) || commitment.validatorSetID == nextValidatorSet.id - 1``` will just work but I'll definetely do more tests.

Reply:
https://github.com/Snowfork/snowbridge/pull/1215/commits/c11ed7d9beefe9fc0ee924c2f3c1ed6a06aa61bc